pkg/client/unversioned/componentstatuses.go

```diff
@@ -22,6 +22,10 @@ import (
 	"k8s.io/kubernetes/pkg/labels"
 )
 
+const (
+	componentStatusResourceName string = "componentStatuses"
```

Comment:
@caesarxuchao - this is a good extension to the generator you're doing (obviously in a separate PR)